tests/user_commands/tests.py

```diff
@@ -401,11 +401,8 @@ def test_subparser_dest_required_args(self):
         self.assertIn("bar", out.getvalue())
 
     def test_subparser_invalid_option(self):
-        if PY314:
-            msg = "invalid choice: 'test' (choose from foo)"
-        else:
-            msg = "invalid choice: 'test' (choose from 'foo')"
-        with self.assertRaisesMessage(CommandError, msg):
+        msg = re.compile(r"invalid choice: 'test' \(choose from '?foo'?\)")
```

Comment:
As far as I'm aware there is not need to pre-compile regexp.
```suggestion
        msg = r"invalid choice: 'test' \(choose from '?foo'?\)"
```